comicboxd/app/gql/query.go

```diff
@@ -45,38 +46,25 @@ func Query(r *http.Request, query string, vars map[string]interface{}, response
 
 	GQLHandler.ServeHTTP(resp, newR)
 
-	qr := graphql.Result{}
+	qr := graphql.Response{}
 
-	json.NewDecoder(resp.Body).Decode(&qr)
-	// resp.Body.Close()
-	for _, err := range qr.Errors {
-		return fmt.Errorf("graphql error: %v", err)
+	err = json.NewDecoder(resp.Body).Decode(&qr)
+	if err != nil {
+		return err
 	}
 
-	data, ok := qr.Data.(map[string]interface{})
-	if !ok {
-		return fmt.Errorf("data not a map")
+	for _, err := range qr.Errors {
+		return fmt.Errorf("graphql error: %v", err)
 	}
 
-	if len(data) > 1 {
-		return fmt.Errorf("more than one return value")
-	}
-	for _, fieldData := range data {
-		jData, err := json.Marshal(fieldData)
-		if err != nil {
-			return err
-		}
-		if response != nil {
-			err = json.Unmarshal(jData, response)
-			if err != nil {
-				return err
-			}
-		}
+	// removes the outer object from the json. I can't think of a nicer way of
```

Comment:
I don't have a better solution right now, but we should make a note to come back and think of something nicer later.